app/src/main/java/org/schabi/newpipe/util/ListHelper.java

```diff
@@ -13,126 +14,72 @@
     }
 
     /**
      * @see #getDefaultResolutionIndex(String, String, MediaFormat, List)
      */
-    public static int getDefaultResolutionIndex(Context context, List<VideoStream> videoStreams, String defaultResolution) {
+    public static int getResolutionIndex(Context context, List<VideoStream> videoStreams, String defaultResolution) {
         return getDefaultResolutionWithDefaultFormat(context, defaultResolution, videoStreams);
     }
 
     /**
      * @see #getDefaultResolutionIndex(String, String, MediaFormat, List)
      */
     public static int getPopupDefaultResolutionIndex(Context context, List<VideoStream> videoStreams) {
-        SharedPreferences defaultPreferences = PreferenceManager.getDefaultSharedPreferences(context);
-        if (defaultPreferences == null) return 0;
-
-        String defaultResolution = defaultPreferences.getString(context.getString(R.string.default_popup_resolution_key), context.getString(R.string.default_popup_resolution_value));
-        return getPopupDefaultResolutionIndex(context, videoStreams, defaultResolution);
+        String defaultResolution = computeDefaultResolution(context,
+                R.string.default_popup_resolution_key, R.string.default_popup_resolution_value);
+        return getDefaultResolutionWithDefaultFormat(context, defaultResolution, videoStreams);
     }
 
     /**
      * @see #getDefaultResolutionIndex(String, String, MediaFormat, List)
      */
-    public static int getPopupDefaultResolutionIndex(Context context, List<VideoStream> videoStreams, String defaultResolution) {
+    public static int getPopupResolutionIndex(Context context, List<VideoStream> videoStreams, String defaultResolution) {
         return getDefaultResolutionWithDefaultFormat(context, defaultResolution, videoStreams);
     }
 
     public static int getDefaultAudioFormat(Context context, List<AudioStream> audioStreams) {
-        MediaFormat defaultFormat = getDefaultFormat(context, R.string.default_audio_format_key, R.string.default_audio_format_value);
-        return getHighestQualityAudioIndex(defaultFormat, audioStreams);
-    }
-
-    public static int getHighestQualityAudioIndex(List<AudioStream> audioStreams) {
-        if (audioStreams == null || audioStreams.isEmpty()) return -1;
-
-        int highestQualityIndex = 0;
-        if (audioStreams.size() > 1) for (int i = 1; i < audioStreams.size(); i++) {
-            AudioStream audioStream = audioStreams.get(i);
-            if (audioStream.getAverageBitrate() >= audioStreams.get(highestQualityIndex).getAverageBitrate()) highestQualityIndex = i;
+        MediaFormat defaultFormat = getDefaultFormat(context, R.string.default_audio_format_key,
+                R.string.default_audio_format_value);
+
+        // If the user has chosen to limit resolution to conserve mobile data
+        // usage then we should also limit our audio usage.
+        int result;
+        if (isLimitingDataUsage(context)) {
```

Comment:
why not
```
if (isLimitingDataUsage(context)) {
   return getMostCompactAudioIndex(defaultFormat, audioStreams);
} else {
   return getHighestQualityAudioIndex(defaultFormat, audioStreams);
}

Reply:
I think because my original changes were more complicated (multiple limiting levels). Some people prefer one return at the end of a method so I kept it. It doesn't matter to me either way, I'll throw in the returns.

Reply:
I am an R-Value/functional fan, so I think return would fit well here :)